internal/pkg/cli/interfaces.go

```diff
@@ -229,3 +233,16 @@ type statusDescriber interface {
 type pipelineGetter interface {
 	GetPipeline(pipelineName string) (*codepipeline.Pipeline, error)
 }
+
+type deleteAppExecutor interface {
+	Execute() error
+}
+
+type deletePipelineRunner interface {
```

Comment:
There's already a runner interface with a different method signature. 

While I appreciate the recommendation of the interface names, I'm not actually a huge fan of it in this scenario, only bc I feel like it makes the interfaces a bit less descriptive for the sake of being possibly prematurely general and in accordance to code style. I also don't love the generalization bc we don't actually implement 'Run' generally in all of our delete*Resource commands, if the idea was to reuse it. 

At least for this interface, I'll keep it like this for now, but let me know if you feel strongly otherwise.

Reply:
ah the way i think we can get around is by having the field names in the struct itself be explicit, for example:
```go
type deleteProjOpts struct {
    ...
    deletePipelineOpts runner
    ...
}
```
Reading `o.deletePipelineOpts.Run()` makes it obvious that we are running the `"pipeline delete"` command so we can still be descriptive about what's `run`ning while keeping the interface flexible like the language's recommendation.

Reply:
Sure, what we name it in the field of the opts struct should certainly be descriptive but in this case I was arguing that it makes it hard to find the implementation, which is already opaque enough as it is (esp for those of us without certain IDE tools ;) ) In any case I was getting errors generating mocks bc of the existing `runner` interface.